Src/AutoNSubstituteUnitTest/SubstituteAttributeRelayTest.cs

```diff
@@ -0,0 +1,164 @@
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteAttributeRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNull()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(null, context);
+            // Verify outcome
+            var expected = new NoSpecimen(null);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotICustomAttributeProvider()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = new object();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var expected = new NoSpecimen(request);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenICustomAttributeProviderDoesNotReturnExpectedAttributeType()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            request.GetCustomAttributes(Arg.Any<Type>(), Arg.Any<bool>()).Returns(new object[0]);
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var expected = new NoSpecimen(request);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateGetsInheritedAttributesFromProviderBecauseAttributesAreInheritedByDefault()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            sut.Create(request, context);
+            // Verify outcome
+            request.Received().GetCustomAttributes(typeof(SubstituteAttribute), true);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsSpecimenResolvedByContextFromSubstituteRequest()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ParameterInfo>();
+            request.ParameterType.Returns(typeof(IInterface));
+            request.GetCustomAttributes(typeof(SubstituteAttribute), Arg.Any<bool>()).Returns(new[] { new SubstituteAttribute() });

```

Comment:
Not really. Using the `Devil's Advocate` technique, I can sabotage the implementation like this:

```C#
public object Create(object request, ISpecimenContext context)
{
    if (context == null)
    {
        throw new ArgumentNullException("context");
    }

    var customAttributeProvider = request as ICustomAttributeProvider;
    if (customAttributeProvider == null)
    {
        return new NoSpecimen(request);
    }

    customAttributeProvider.GetCustomAttributes(typeof(SubstituteAttribute), true);
    var attribute = customAttributeProvider.GetCustomAttributes(typeof(SubstituteAttribute), false)
            .OfType<SubstituteAttribute>().FirstOrDefault();
    if (attribute == null)
    {
        return new NoSpecimen(request);
    }

    object substituteRequest = CreateSubstituteRequest(customAttributeProvider, attribute);
    return context.Resolve(substituteRequest);
}
```

Not that I would, but the `Devil's Advocate` is a review technique to demonstrate that the test may not be testing what one thinks it's testing.

With that above incorrect implementation, all tests still pass.